.github/workflows/golang-build-test.yml

```diff
@@ -0,0 +1,23 @@
+name: golang-build-test
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.21.x'
+      - name: Install dependencies
+        run: go get .
```

Comment:
we could probably leverage GA cache

Reply:
it looks like that the latest version of setup-go will automatically cache the dependencies
https://github.blog/changelog/2023-03-24-github-actions-the-setup-go-action-now-enables-caching-by-default/